### ISSUE TITLE
feat: kendo-tabstrip-tab > custom per tab css classes

### ISSUE THREAD

**Issue by cluetjen** [telerik/kendo-angular#1739]
## I'm submitting a...
  * Feature patch file (requests to our UserVoice Portal at https://goo.gl/HsVr7R)

… but cannot use the feedback portal because there's no option to attach files. Ok here?

## Current behavior
There seems to be no (easy) way to attach a custom css class to a specific kendo-tabstrip-tab. The following example is NOT working (note kendo-tabstrip-tab > class):

```
      <kendo-tabstrip (tabSelect)="onSelect($event)" [ngStyle]="{'width': '400px'}" [animate]="true" tabPosition="left">
        <kendo-tabstrip-tab [title]="'Paris'" [selected]="true" [class]="{'my-additional-class':true}">
            <ng-template kendoTabContent>
                  <h3>Content 1</h3>
            </ng-template>
        </kendo-tabstrip-tab>
```

## Expected behavior

Would be cool if we could add custom classes to specific tabs.

Example case: We've a form that spans over multiple tabs. On submit we want to check if form values are valid and highlight invalid ones. Problem: User will not see most of the form controls because they are on inactive tabs. Solution: Let's mark the tab title as "invalid" (= add a custom class and be happy).

The following patch adds support for "class" attribute (as used in other kendo components) to kendo-tabstrip-tab component:

[0001-feat-tabstrip-class-attribute-added.zip](https://github.com/telerik/kendo-angular/files/2294601/0001-feat-tabstrip-class-attribute-added.zip)

```
From abcd72a76e66e4d7602d20e3d8e3436d8cc0c375 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christoph=20L=C3=BCtjen?= <info@christoph-luetjen.de>
Date: Thu, 16 Aug 2018 19:10:56 +0200
Subject: [PATCH] feat(tabstrip): class attribute added.

---
 examples/tabstrip_basic.ts             | 2 +-
 src/tabstrip/tabstrip-tab.component.ts | 7 +++++++
 src/tabstrip/tabstrip.component.ts     | 1 +
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/examples/tabstrip_basic.ts b/examples/tabstrip_basic.ts
index e9e318d..1e34a70 100644
--- a/examples/tabstrip_basic.ts
+++ b/examples/tabstrip_basic.ts
@@ -7,7 +7,7 @@ import { LayoutModule } from '@progress/kendo-angular-layout';
   selector: 'my-app',
   template: `
       <kendo-tabstrip (tabSelect)="onSelect($event)" [ngStyle]="{'width': '400px'}" [animate]="true" tabPosition="left">
-        <kendo-tabstrip-tab [title]="'Paris'" [selected]="true">
+        <kendo-tabstrip-tab [title]="'Paris'" [selected]="true" [class]="{'my-additional-class':true}">
             <ng-template kendoTabContent>
                   <h3>Content 1</h3>
             </ng-template>
diff --git a/src/tabstrip/tabstrip-tab.component.ts b/src/tabstrip/tabstrip-tab.component.ts
index 84d091a..6dd75ff 100644
--- a/src/tabstrip/tabstrip-tab.component.ts
+++ b/src/tabstrip/tabstrip-tab.component.ts
@@ -37,6 +37,13 @@ export class TabStripTabComponent implements AfterContentInit, OnChanges {
      */
     @Input() public selected: boolean;
 
+    /**
+     * Sets custom CSS classes. Under the hood, to apply the property,
+     * the `class` option uses the
+     * [NgClass](https://angular.io/docs/ts/latest/api/common/index/NgClass-directive.html) directive.
+     */
+    @Input() public class: string | string[] | Set<string> | { [key: string]: any };
+
     public get tabContent(): TabContentDirective {
         return this._tabContent.first;
     }
diff --git a/src/tabstrip/tabstrip.component.ts b/src/tabstrip/tabstrip.component.ts
index 0042a0c..5443afe 100644
--- a/src/tabstrip/tabstrip.component.ts
+++ b/src/tabstrip/tabstrip.component.ts
@@ -77,6 +77,7 @@ import { Subscription } from 'rxjs/Subscription';
                 <li *ngFor="let tab of tabs; let i = index;" (click)="selectTab(i)"
                     role="tab"
                     [id]="tabId(i)"
+                    [ngClass]="tab.class"
                     [class.k-item]="true"
                     [class.k-state-default]="true"
                     [class.k-state-active]="tab.active"
-- 
2.17.1.windows.2


```

**Comment by ipeshev** [comment]
Feature available in [latest dev version](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/).

**Comment by danielkaradachki** [comment]
Released as cssClass in 3.3.0

https://stackblitz.com/edit/angular-uo6uq1?file=app/app.component.ts
